pep-0572.rst

```diff
@@ -359,6 +359,127 @@ found in assignment statements:
+|   tz = self._tzstr()                                |   if tz := self._tzstr():                           |
+|   if tz:                                            |       s += tz                                       |
+|       s += tz                                       |   return s                                          |
+|   return s                                          |                                                     |
+|                                                     |                                                     |
++-----------------------------------------------------+-----------------------------------------------------+
+
+*tz* is only used for ``s += tz``, moving its assignment inside the if
+helps to show its scope.
+
+sysconfig.py
+^^^^^^^^^^^^
+
++------------------------------------+-----------------------------------------+
+| Current code                       | Replaced with                           |
++------------------------------------+-----------------------------------------+
+|                                    |                                         |
+| ::                                 | ::                                      |
+|                                    |                                         |
+|   while True:                      |   while line := fp.readline():          |
+|       line = fp.readline()         |       if m := define_rx.match(line):    |
+|       if not line:                 |           n, v = m.group(1, 2)          |
+|           break                    |           try:                          |
+|       m = define_rx.match(line)    |               v = int(v)                |
+|       if m:                        |           except ValueError:            |
+|           n, v = m.group(1, 2)     |               pass                      |
+|           try:                     |           vars[n] = v                   |
+|               v = int(v)           |       elif m := undef_rx.match(line):   |
+|           except ValueError:       |           vars[m.group(1)] = 0          |
+|               pass                 |                                         |
+|           vars[n] = v              |                                         |
+|       else:                        |                                         |
+|           m = undef_rx.match(line) |                                         |
+|           if m:                    |                                         |
+|               vars[m.group(1)] = 0 |                                         |
+|                                    |                                         |
++------------------------------------+-----------------------------------------+
+
+Calling ``fp.readline()`` in the ``while`` condition and calling
+``.match()`` on the if lines make the code more compact without making
+it harder to understand.
+
```

Comment:
Just leaving a note of appreciation for this example - I find the right-side version significantly easier to follow!